conda/recipes/cuml/meta.yaml

```diff
@@ -33,40 +33,54 @@ build:
     - SCCACHE_S3_KEY_PREFIX=cuml-linux64 # [linux64]
     - SCCACHE_S3_USE_SSL
   ignore_run_exports_from:
-    - {{ compiler('cuda') }}
+    {% if cuda_major == "11" %}
+    - {{ compiler('cuda11') }}
+    {% endif %}
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }} {{ cuda_version }}
+    {% if cuda_major == "11" %}
+    - {{ compiler('cuda11') }} ={{ cuda_version }}
+    {% else %}
+    - {{ compiler('cuda') }}
+    {% endif %}
+    - cuda-version ={{ cuda_version }}
     - cmake {{ cmake_version }}
     - ninja
     - sysroot_{{ target_platform }} {{ sysroot_version }}
   host:
-    - cuda-python >=11.7.1,<12.0
-    - cudatoolkit ={{ cuda_version }}
+    - cuda-version ={{ cuda_version }}
+    {% if cuda_major == "11" %}
+    - cuda-python ==11.7.1
+    - cudatoolkit
+    - libcublas {{ cuda11_libcublas_host_version }}
+    - libcurand {{ cuda11_libcurand_host_version }}
+    - libcusolver {{ cuda11_libcusolver_host_version }}
+    - libcusparse {{ cuda11_libcusparse_host_version }}
```

Comment:
Do any of these CUDA math libraries need to be in `host` for cuml Python? Are they directly used by cuml Python or only indirectly through libcuml? If the latter, maybe we can get away with only specifying them in libcuml's recipe.

I wonder if these (non-dev) libraries are also remnants of the changes used when migrating to GitHub Actions before RAFT recipes were complete. It would surprise me if these are used directly in cuml Python, since they're not mentioned in `python/CMakeLists.txt`. However, I see that there are linkages in the installed packages: for example, `ldd manifold/umap.cpytho-310-x86_64-linux-gnu.so` shows `libcufft.so.10`, `libcurand.so.10`, `libcusolver.so.11`, `libcublas.so.11`, `libcusparse.so.11`.

Reply:
After thinking about this more, we probably want to remove these. Putting “runtime” (non-dev) libraries in the `host` section probably does nothing, except perhaps satisfy conda linking checks (which we can verify by looking at the warnings).